client/state/ui/section/utils.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Internal dependencies
+ */
+import { createReducer } from 'state/utils';
+import { SECTION_SET } from 'state/action-types';
+
+/**
+ * Given an initial state and section object key to manage, returns a reducer
+ * function which manages state for the section key.
+ *
+ * @param  {*}        initialState Initial state value
+ * @param  {String}   key          Section object key
+ * @return {Function}              Section reducer
+ */
+export function createSectionReducer( initialState, key ) {
+	return createReducer( initialState, {
+		[ SECTION_SET ]: ( state, { section } ) => {
+			if ( section && section.hasOwnProperty( key ) ) {
+				return section[ key ];
+			}
+
+			return state;
```

Comment:
one pattern I have been doing recently for the sake of immutability and shallow reference checks is to return the previous state if the updating action contains the same value. otherwise we trick the app into thinking there's a new value when there really isn't.

obviously if this is a bool, string, or number this doesn't apply…

```js
if (
	section &&
	section.hasOwnProperty( key ) &&
	state !== section[ key ]
) {
	return section[ key ]
}
```